pkg/operator/csidriveroperator/crcontroller.go

```diff
@@ -217,10 +217,6 @@ func (c *CSIDriverOperatorCRController) syncConditions(ctx context.Context, cond
 	}
 
 	upgradeableCond := status.UnionCondition(operatorapi.OperatorStatusTypeUpgradeable, operatorapi.ConditionTrue, nil, conditions...)
-	upgradeableCond.Type = c.crConditionName(operatorapi.OperatorStatusTypeUpgradeable)
-	if upgradeableCond.Status == operatorapi.ConditionUnknown {
-		upgradeableCond.Status = operatorapi.ConditionTrue
-	}
```

Comment:
I think it is correct. At least it worked for AWS. With this PR, OCP can't be installed:

```
Some cluster operators are not ready: storage (Upgradeable=Unknown _NoData: )
```

Reply:
hmm, I think this is because the Union code in library-go adds "Condition: Unknown" if that condition type is not there in the CR's condition at all. May be we should only union if `Condition: Unknown` is present.